### PR TITLE
Added paths filter to GitHub actions

### LINKED ISSUE #203: Make actions to run only on changed files.
### Is your enhancement related to a problem? Please describe.

To reduce pipeline resources and skip unnecessary checks on PR (consider running E2E tests for minor readme file changes), I think it is better to run actions only when specific files are changed.

Inspired from https://github.com/10up/restricted-site-access/pull/227

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by iamdharmesh**
### Description of the Change
This PR adds a paths filter in GitHub action workflows to run it only on specific file extensions. For example, This PR doesn't contain any PHP, JS, or CSS file changes. So, the related workflows(E2E tests, PHPUnit, PHPCS, PHP Compatibility, and ESLint) will not be run against this PR. 

Closes #203 

### How to test the Change
See GitHub Actions check on this PR, E2E tests, PHPUnit, PHPCS, PHP Compatibility, and ESLint check should not run on this PR as we made changes in workflow files only.

### Credits
Props @iamdharmesh

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Review by peterwilsoncc — DISMISSED** [review]
Fake approval to see what happens.

**Review by peterwilsoncc — DISMISSED** [review]
Another fake approval to test things now that develop has been merged in.

**Review by peterwilsoncc — CHANGES_REQUESTED** [review]
Sorry for the review spam while I was testing the effects of this PR.

In its current state, it prevents merging of approved PRs as the required actions have not completed.

I think what you'll want to do is something like this (pseudo code)

```yaml
step:
  Name: Succeed early if files are not modified
  run: |
    if ( no_relevant_file_modified() ) {
      finish_action_with_success();
    }
```

I'm not sure if that is possible though, it's something you'll need to check in the docs.

Screen Shot 2022-12-01 at 1 41 46 pm

**Comment by iamdharmesh** [comment]
@peterwilsoncc we can handle this by adding a generic workflow that ignores the path we added to the actual workflow So, it will run instead of the actual workflow and we can satisfy the `required` condition. More info can be found [here](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks)

I have created and added that workflow for PHPCS([c4d298d](https://github.com/10up/autoshare-for-twitter/pull/204/commits/c4d298dade8f3527dbe9cfd9752544df38cdadfe)) and it works as expected. what do you think about this approach? If you think adding an additional dummy workflow for each required workflow is additional overhead then we might add path filters only on heavy workflows like E2E tests OR maybe adjust required workflow settings.

Thanks
cc: @dkotter

**Comment by peterwilsoncc** [comment]
Thank @iamdharmesh -- that looks like a really nice approach. 

It does require repeating the paths in two files, so would need comments to alert people to change them in both locations. You could use a constant but I think that for two instances that would be over-engineering.

File: .github/workflows/ignore-eslint.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    branches:
+      - develop
+    paths-ignore:
+      - '**.js'
+
+jobs:
+  eslint:
+    runs-on: ubuntu-latest
+    steps:
+      - run: 'echo "No ESLint check required"'
```
Comment (peterwilsoncc) [inline .github/workflows/ignore-eslint.yml]: Nit: EOF line break would be lovely.

File: .github/workflows/ignore-phpunit.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  phpunit:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.4', '8.0']
+    steps:
+      - run: 'echo "No PHPUnit check required"'
```
Comment (peterwilsoncc) [inline .github/workflows/ignore-phpunit.yml]:
```suggestion
      - run: 'echo "No PHPUnit check required"'

```

Nit: EOF line break would be lovely

**Review by peterwilsoncc — DISMISSED** [review]
LGTM with a minor nitpick inline.

I tested the changes with #209 to ensure that the tests run and fail as expected when needed.

**Review by peterwilsoncc — APPROVED** [review]
Thank you! LGTM.

**Pull request merged**